File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int area = brown + yellow;
        for (int height = 3; height < (int) Math.sqrt(area) + 1; height++) {
            if (area % height != 0) {
                continue;
            }
            int width = area / height;
            if (area - (width - 2) * (height - 2) == brown) {
                return new int[] {width, height};
            }
        }
        return null;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    for h in range(3, int((brown + yellow) ** 0.5 + 1)):
        if (brown + yellow) % h == 0:
            w = (brown + yellow) // h
            if w * h - (w - 2) * (h - 2) == brown:
                return [w, h]